addons/calendar/models/calendar_attendee.py

```diff
@@ -109,12 +109,36 @@ def _send_mail_to_attendees(self, mail_template, force_send=False):
         # get ics file for all meetings
         ics_files = self.mapped('event_id')._get_ics_file()
 
+        # If Mail template has an attachment, add it to every attendee's mail. Why? Cause it need to be attached to mail.compose.message to be send along the message.
+        # Batching coping of the atachments to improve performance (avoids copy in the loop)
+        if mail_template.attachment_ids:
+
+            # Prepare attachments values for each attendee (the same values for each attendee)
+            attachments_values = mail_template.attachment_ids.copy_data({'res_id': 0, 'res_model': 'mail.compose.message'})
+            attachments_values *= len(self)
+            attachment_for_all_attendees_ids = self.env['ir.attachment'].create(attachments_values).ids
+
+            # Mail template can contain multiple attachments, that's why here we have to reshape list of attachments into per-attendee groups
+            #  [ attach1_for_attendee1, attach2_for_attendee1, attach1_for_attendee2, attach2_for_attendee2]
+            # is reshaped into per-attendee mapping:
+            # {
+            #   attendee1: [attach1_for_attendee1, attach2_for_attendee1]
+            #   attendee2: [attach1_for_attendee2, attach2_for_attendee2]
+            # }
+            number_of_attachments_per_attende = len(mail_template.attachment_ids)
+            attendee_id_attachment_id_map = {}
+            for i, attendee_id in enumerate(self.ids):
+                attendee_attachments_ids = attachment_for_all_attendees_ids[i:i + number_of_attachments_per_attende]
+                attendee_id_attachment_id_map[attendee_id] = attendee_attachments_ids
```

Comment:
In fact you can still make this with a dict comprehension I think
```
{
    attendee: attendee_attachment_ids[n * template_attachment_count: n * (template_attachment_count + 1)]
    for n, attendee in numerate(attendee)
}
```
It's usually more efficient, I think. And I'd argue more readable.

Would also be fine not to have a dict and to get the attachments in the next loop as they're all the same anyway.